### PR TITLE
Replace EnumValidator - Unit Tests and Changing the Validator

### LINKED ISSUE #4639: Remove remaining `EnumValidator`
An addendum to https://github.com/dotnet/winforms/pull/4043 that replaced enum validations with source generators. We still have runtime enum validations lurking around, and they needs to be cleaned out and replaced with the compile-time validations.
https://github.com/dotnet/winforms/blob/2178e7a66ed27332202659c3ea86649dc4578c3d/src/System.Windows.Forms/src/System/Windows/Forms/WinFormsUtils.cs#L334-L377

### PR DESCRIPTION
**Pull request by wreuel**
Fixes #4639

## Proposed changes

- Created tests to Label and RadioButton
- Remove Enum Validator on class

## Customer Impact

- None

## Regression? 

- No

## Risk

- None

## Test methodology 

- Unit Tests created before change

- 

.NET SDK (reflecting any global.json):
Version: 6.0.100-preview.2.21155.3
Commit: 1a9103db2d

Runtime Environment:
OS Name: Windows
OS Version: 10.0.19042
OS Platform: Windows
RID: win10-x64
Base Path: C:\Program Files\dotnet\sdk\6.0.100-preview.2.21155.3\

Host (useful for support):
Version: 6.0.0-preview.2.21154.6
Commit: 3eaf1f316b

.NET SDKs installed:
3.1.407 [C:\Program Files\dotnet\sdk]
5.0.104 [C:\Program Files\dotnet\sdk]
5.0.201 [C:\Program Files\dotnet\sdk]
6.0.100-preview.2.21155.3 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
Microsoft.AspNetCore.All 2.1.26 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
Microsoft.AspNetCore.App 2.1.26 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 3.1.13 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 5.0.4 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 6.0.0-preview.2.21154.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.NETCore.App 2.1.26 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 3.1.6 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 3.1.13 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 5.0.4 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 6.0.0-preview.2.21154.6 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.WindowsDesktop.App 3.1.6 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
Microsoft.WindowsDesktop.App 3.1.13 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
Microsoft.WindowsDesktop.App 5.0.4 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
Microsoft.WindowsDesktop.App 6.0.0-preview.2.21154.2 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/4782)

### CODE REVIEW

**Comment by wreuel** [comment]
@RussKie I couldn't do the squash to my 4 PR Merge (To Update my Fork), I did get it why, but I did for a new with all the code done. If you can recommend a doc or video on how to do the squash or on how to update my Fork, I would be gratefull.

**Comment by RussKie** [comment]
There was no reason to close the other PR 😉 
Please have a look at [these guidelines](https://github.com/gitextensions/gitextensions/wiki/How-To%3A-Squash-and-Rebase-your-changes), hopefully you'd find them helpful.

**Comment by RussKie** [comment]
image

You still have 4 unrelated commits in the PR. The easiest way to remove those is to perform an _interactive rebase_ on top of the latest main branch, _dropping_ the unrelated commits.
It would look something like this:
1. Fetch the latest main, e.g. `git fetch --progress "remote" +refs/heads/main`
2. Rebase on to the latest main, e.g. `git -c rebase.autoSquash=false rebase -i --autosquash <sha of the latest main HEAD>`
3. In the dialog drop irrelevant commits, e.g.
image
4. Force-push back into your fork.

**Comment by wreuel** [comment]
Sorry my delay @RussKie, but now I think is good to go!

**Comment by AraHaan** [comment]
After thinking about it should this get merged when #4809 is done? I will try to get it done as soon as I fix #4751 so then it will also unblock #4746.

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/RadioButtonTests.cs [hunk 1]
```diff
@@ -361,6 +361,22 @@ public void RadioRadioButton_TextAlign_SetInvalidValue_ThrowsInvalidEnumArgument
             using var control = new SubRadioButton();
             Assert.Throws<InvalidEnumArgumentException>("value", () => control.TextAlign = value);
         }
+		
+		[WinFormsTheory]
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/RadioButtonTests.cs]: Please use spaces
image

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/RadioButtonTests.cs [hunk 2]
```diff
@@ -361,6 +361,22 @@ public void RadioRadioButton_TextAlign_SetInvalidValue_ThrowsInvalidEnumArgument
...
+            Assert.Throws<InvalidEnumArgumentException>("value", () => control.CheckAlign = value);
+        }
+
+        [WinFormsTheory]
+        [CommonMemberData(nameof(CommonTestHelper.GetEnumTypeTheoryDataInvalid), typeof(ContentAlignment))]
+        public void RadioRadioButton_CheckAlign_SetInvalidValue_ThrowsInvalidEnumArgumentException(ContentAlignment value)
+        {
+            using var control = new SubRadioButton();
+            Assert.Throws<InvalidEnumArgumentException>("value", () => control.CheckAlign = value);
+        }
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/RadioButtonTests.cs:379]: What's the difference between these two tests (besides the name)?

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/RadioButtonTests.cs [hunk 3]
```diff
@@ -361,8 +361,8 @@ public void RadioRadioButton_TextAlign_SetInvalidValue_ThrowsInvalidEnumArgument
             using var control = new SubRadioButton();
             Assert.Throws<InvalidEnumArgumentException>("value", () => control.TextAlign = value);
         }
-		
-		[WinFormsTheory]
+
+        [WinFormsTheory]
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/RadioButtonTests.cs]: This should be fixed in the previous commit (hint: `git fixup` 😉)

**Review by RussKie — COMMENTED** [review]
👍 
Few comments

**Comment by RussKie** [comment]
@wreuel a friendly nudge

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/RadioButtonTests.cs (hunk 2 above)
Comment (AraHaan) [reply src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/RadioButtonTests.cs:379]: I do not think there is a difference at all. tbh on this test cant it be merged into a single test that can take in an generic type parameter that it then news up for ``control``?

**Comment by wreuel** [comment]
@RussKie got a problem with my pc, and in the end I forgot this, gonna fix this week! 😃

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/RadioButtonTests.cs (hunk 2 above)
Comment (wreuel) [reply src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/RadioButtonTests.cs:379]: @RussKie  and @AraHaan 

Ok, I fixed the first one but did not commit yet, but how should I fixed this, can I do something like pass the object with an InlineData and change the method signature to 

` public void RadioRadioButton_CheckAlign_SetInvalidValue_ThrowsInvalidEnumArgumentException(ContentAlignment value, Control control)` ? If so, how come?

Sorry, got a little confused here, or should I just delete the test from line 366 to 371 ?

Comment (RussKie) [reply src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/RadioButtonTests.cs:379]: The question here is what's the reason to test against a `RadioButton` and a `SubRadioButton`? If there's none - then remove lines:373-380.

**Pull request merged**